### PR TITLE
Add Media-Type response headers to API responses

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR completes out the set of tasks for configuring Pyramid to handle accept header content negotiation, and, as such, support the technical versioning of our API.

This last piece adds a response header to all successful (non-error) responses from API endpoints (views). The value of that response header, `X-Hypothesis-Media-Type`, indicates which version of the API serviced the request, so clients can confirm that they got the version they expected. This approach is cribbed from [github's REST API](https://developer.github.com/v3/#current-version), with a small difference in that I chose to represent the entire media type in its value, versus a truncated version.

I prefer `X-` prefixed custom headers because they're much, much easier to spot and identify immediately as custom headers. But argue against me if you will!

The current logic for what to set the header value to is static/dumb. Expanding that logic is part of adding `"v2"` to the app's list of known API versions.

Fixes https://github.com/hypothesis/product-backlog/issues/941

### CODE REVIEW

**Comment by robertknight** [comment]
> so clients can confirm that they got the version they expected. 

What would a client do with this header? Is it purely there as a debugging/informational aid or are there other uses?

**Comment by robertknight** [comment]
> I prefer X- prefixed custom headers because they're much, much easier to spot and identify immediately as custom headers. But argue against me if you will!

I think that above all else we should be consistent, so either we prefix all of our custom headers or none of them.

My initial reaction is that the `X-` and `Hypothesis-` prefixes here are both serving the same purpose, to indicate that the format and purpose of this header is defined by our application rather than by an RFC or other web spec, except that `Hypothesis-` is more specific. This is only a weakly-held opinion.

**Comment by lyzadanger** [comment]
> > so clients can confirm that they got the version they expected.
> 
> What would a client do with this header? Is it purely there as a debugging/informational aid or are there other uses?

Primarily as a debug and informational aid, as you mentioned. Could help us with support requests, also, and our own internal debug (i.e. not just clients benefit). Seems useful, but are you thinking it doesn't? Because I could totally not do this :). I was following GitHub API's lead; it seemed like a nice-to-have little feature...

**Comment by lyzadanger** [comment]
> > I prefer X- prefixed custom headers because they're much, much easier to spot and identify immediately as custom headers. But argue against me if you will!
> 
> I think that above all else we should be consistent, so either we prefix all of our custom headers or none of them.
> 
> My initial reaction is that the `X-` and `Hypothesis-` prefixes here are both serving the same purpose, to indicate that the format and purpose of this header is defined by our application rather than by an RFC or other web spec, except that `Hypothesis-` is more specific. This is only a weakly-held opinion.

For us, yes—we'd immediately spot `Hypothesis` and know it was custom. Would it be as immediately obvious to clients/people who aren't steeped all day in Hypothesis? Maybe so! I'm just musing here. Your point is a good one.

File: h/views/api/helpers/media_types.py [hunk 1]
```diff
@@ -31,13 +31,23 @@ def valid_media_types(versions=None):
...
 
-    :arg versions: List of version strings to include in the valid media types.
-                   Defaults to the app's known version list.
+    :rtype: list(str)
+    """
+    valid_types = ["*/*", "application/json"] + version_media_types()
+    return valid_types
+
+
+def version_media_types(versions=None):
```
Comment (robertknight) [inline h/views/api/helpers/media_types.py]: Do you envision this helper being used outside of this module? Currently it looks like it is just a helper for `valid_media_type`.

**Pull request merged**